go/p2p/peermgmt/peermgr.go

```diff
@@ -188,15 +188,23 @@ func (m *PeerManager) RegisterProtocol(p core.ProtocolID, minPeers int, totalPee
 	}
 
 	m.protocols[p] = &watermark{minPeers, totalPeers}
-	m.discovery.startAdvertising(string(p))
-
 	m.logger.Debug("protocol registered",
 		"protocol", p,
 		"min_peers", minPeers,
 		"total_peers", totalPeers,
 	)
 }
 
+// AdvertiseProtocol starts advertising readiness to serve the specified protocol.
+//
+// This enables remote peers without existing connection to find the host node.
+func (m *PeerManager) AdvertiseProtocol(p core.ProtocolID) {
```

Comment:
You should do the same for topics, i.e. you should also remove `startAdvertising` from `RegisterTopic`.

Reply:
The discovery and peer manager also support protocol removal, although that’s not needed in our case (not sure why I included it earlier, maybe just for completeness). Previously, registering protocols and topics would both start advertising and initiate node discovery. Now, these actions are split into two separate methods: one for discovery and one for advertising. Therefore, if you modify the registration logic, you’ll also need to update the unregistration accordingly. It might also be helpful to consider renaming the `Register*` methods to reflect this change. Yes, it is annoying, but you should be consistent. You could also remove support for protocol removal (not sure if this is good, though).